### PR TITLE
Add multi window example

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes gtk-rs/gtk/issues/484.

### CODE REVIEW

File: src/multi_windows.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    });
+
+    window.show_all();
+    window
+}
+
+fn generate_new_id(windows: Rc<RefCell<HashMap<usize, gtk::Window>>>) -> usize {
+    let mut len = {
+        windows.borrow().values().count()
+    };
```
Comment (EPashkin) [inline src/multi_windows.rs]: IMHO unneeded {}

**Comment by EPashkin** [comment]
Recreate with pending branch please

File: src/multi_windows.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/multi_windows.rs]: Actually no, it's not. Good point. Just setting to 0 is fine.

Comment (EPashkin) [inline src/multi_windows.rs]: If you set 0, then will be unneeded loop.

**Comment by EPashkin** [comment]
I don't know that editing can change branch.
@GuillaumeGomez, can you also change others PRs?

File: src/multi_windows.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/multi_windows.rs]: The loop is still needed because I need every subwindow to have a different id.

**Comment by GuillaumeGomez** [comment]
Dark magic! :p

And sure.

**Comment by EPashkin** [comment]
May be we really need change branch setup for this crate to master/production

**Comment by GuillaumeGomez** [comment]
Instead of doing this, I think we should maybe add a README note about PRs or maybe a PR message (since github allows it...).

**Comment by EPashkin** [comment]
It already added https://github.com/gtk-rs/examples/blob/master/CONTRIBUTING.md and shown only when attempt to add PR on master. Just not effective.

**Comment by GuillaumeGomez** [comment]
Well, we tried. :p

**Comment by GuillaumeGomez** [comment]
@EPashkin: Updated.

**Comment by EPashkin** [comment]
@GuillaumeGomez file not in Bin

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/bin/multi_windows.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        gtk::main_quit();
+        Inhibit(false)
+    });
+
+    window.show_all();
+    window
+}
+
+fn generate_new_id(windows: Rc<RefCell<HashMap<usize, gtk::Window>>>) -> usize {
+    let mut len = 0;
```
Comment (EPashkin) [inline src/bin/multi_windows.rs]: Maybe it `id` now?

Comment (GuillaumeGomez) [inline src/bin/multi_windows.rs]: Fixed.

File: src/bin/multi_windows.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    let button = gtk::Button::new_with_label("Create new window");
+    button.connect_clicked(move |_| {
+        let new_id = generate_new_id(windows.clone());
+        let new_window = create_sub_window(&windows_title_entry_clone.get_buffer().get_text(),
+                                           main_entry.clone(),
+                                           new_id,
+                                           windows.clone());
+        // Once the new window has been created, we put it into our hashmap so we can update its
+        // title when needed.
+        windows.borrow_mut().insert(new_id, new_window);
```
Comment (EPashkin) [inline src/bin/multi_windows.rs]: May be this better go to end of `create_sub_window` this way all logic will be almost in one place

Comment (GuillaumeGomez) [inline src/bin/multi_windows.rs]: Hum, good idea.

Comment (GuillaumeGomez) [inline src/bin/multi_windows.rs]: Updated.

**Comment by EPashkin** [comment]
Thanks

**Pull request merged**